notifier/notifier.go

```diff
@@ -640,6 +641,17 @@ func newAlertmanagerSet(cfg *config.AlertmanagerConfig, logger log.Logger, metri
 	if err != nil {
 		return nil, err
 	}
+	t := client.Transport
+
+	if cfg.SigV4Config != nil {
+		t, err = sigv4.NewSigV4RoundTripper(cfg.SigV4Config, client.Transport)
+		if err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
In the remote write client, this is also where prometheus supports injecting arbitrary headers from the config:

https://github.com/prometheus/prometheus/blob/db816ce9dbf2768ed2db1619fb2a8b9f9addeae8/storage/remote/client.go#L158-L160

Is that desirable here as well? If for nothing else other than consistency?

Reply:
Limiting to sigv4 per review feedback